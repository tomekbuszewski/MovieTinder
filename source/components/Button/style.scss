@import '../App/globals';

$Button: '.Button';

#{$Button} {
  width: 100%;
  height: 5rem;
  border: 0;
  background: none;
  border-radius: 2.5rem;
  color: $PotPourri;
  cursor: pointer;
  font-family: $fontSecondary;
  font-size: map_get($fonts, text);
  font-weight: 800;
  line-height: 4.5rem;
  outline: 0;
  text-align: center;

  &--Accept {
    background: linear-gradient(45deg, $Matisse 0%, $Bossanova 100%);
  }
  &--Decline {
    width: 14rem;
    border: 0.25rem solid $Bossanova;
    margin-right: 2rem;
  }

  @media (min-width: map_get($mq, md)) {

    &--Decline {
      width: 100%;
      order: 1;
      margin: 2rem 0 0;
    }
  }
}
