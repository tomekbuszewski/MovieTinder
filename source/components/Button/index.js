// @flow
import React from 'react';
import PropTypes from 'prop-types';
import { BUTTON_ACCEPT } from '../../constants/buttonTypes';

import ClassNameBuilder from 'bem-classname-builder';

import style from './style.scss';

const bem = new ClassNameBuilder(style, 'Button', true);

const Button = ({ children, type, onClick, additionalClasses }: { children: String, type: String, onClick: Function, additionalClasses: String }) => {
  const m = type === BUTTON_ACCEPT ? 'Accept' : 'Decline';
  const className = `${bem.get()} ${bem.get({ m })} ${additionalClasses}`;

  return <button onClick={onClick} className={className}>{children}</button>;
};

Button.propTypes = {
  text: PropTypes.string,
  type: PropTypes.string,

};

Button.defaultProps = {
  type: BUTTON_ACCEPT,
  onClick: () => {},
  additionalClasses: ''
};

export default Button;
