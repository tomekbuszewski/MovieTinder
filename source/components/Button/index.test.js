import React from 'react';
import renderer from 'react-test-renderer';

import Button from './index';

describe ('Button tests', () => {
  it ('should match snapshot', () => {
    const tree = renderer.create(<Button type="null">Test</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
