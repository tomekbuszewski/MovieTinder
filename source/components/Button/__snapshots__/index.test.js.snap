// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests should match snapshot 1`] = `
<button
  className="  "
  onClick={[Function]}
>
  Test
</button>
`;
